test/Uri.js

```diff
@@ -48,8 +48,19 @@ describe('Uri', function() {
 		assert.strictEqual('', uri.getOrigin());
 	});
 
+	it('should set procotol http: on localhost uri', function() {
+		var uri = new Uri('localhost:8080');
+		assert.strictEqual('http:', uri.getProtocol());
+		assert.strictEqual('8080', uri.getPort());
+	});
+
+	it('should support all types of uri schemes', function() {
+		var uri = new Uri('test:hostname');
+		assert.strictEqual('test:', uri.getProtocol());
+	});
+
 	it('should support set port on uri', function() {
-		var uri = new Uri('hostname:8080');
```

Comment:
Hey @kienD,

I did a quick research and I figured out that the `uri scheme` accepts, besides letters, digits, plus (+), period (.), or hyphen (-). So, the current validation will not recognise `iris.xpc` as a valid protocol.

Here are some links:
[https://en.wikipedia.org/wiki/Uniform_Resource_Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)
[https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml)